index.js

```diff
@@ -37,7 +37,7 @@ function compose (middleware) {
     function dispatch (i) {
       if (i <= index) return Promise.reject(new Error('next() called multiple times'))
       index = i
-      const fn = middleware[i] || next
+      const fn = wrap(middleware[i]) || next
```

Comment:
The performant solution would be this:

```js
Object.defineProperty('_wrappers', {
  set: wrappers => middleware = wrappers.reduce((mw, wrapper) => wrapper(mw), middeware),
  get: []
})
```

That would be a breaking change though, even if we properly implement get as setting wrappers multiple times would wrap the middleware multiple times. That could also be changed with an if statement of course, but I think we should choose a style to implement it with and keep it consistent.